### PR TITLE
acme mvp test

### PR DESCRIPTION
**Pull request by dpuniaredhat**
Adding automation to Deploy acme container on podman and openshift

1. Update existing acme test cases and add test cases
2. Adding test to deploy acme using podman with Persistent Database and run test cases
3. Adding acme deployment with nss_issuer
4. Adding test to delopy acme on openshift with Persistent Database

Signed-off-by: Deepak Punia dpunia@redhat.com

### CODE REVIEW

File: tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme.yml [hunk 1]
```diff
@@ -9,58 +9,21 @@
...
-   replace: 'bindPassword={{variable.LDAP_PASSWD}}'
+- name: Configuring ACME Responder PKI Issuer
+  shell: pki-server acme-issuer-mod --type pki -Dusername={{variable.CA_ADMIN_USERNAME}} -Dpassword={{variable.CA_PASSWORD}} --instance {{topology}}-CA
+  when: acme_nss_issuer == "false"
 
-- name: Removing existing ldap entry
-  command: sed -i 1,3d "{{variable.ACME_CONFIG_PATH}}/database/ldap/create.ldif"
+- name: Configuring ACME Responder NSS Issuer
+  shell: pki-server acme-issuer-mod --type nss -Dnickname="caSigningCert cert-{{topology}}-CA CA" --instance {{topology}}-CA
+  when: acme_nss_issuer == "true"
```
[the remark this answers is not in the document]
Comment (dpuniaredhat) [reply tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme.yml:23]: Thanks, Will update the logic in future accordingly.

**Comment by dpuniaredhat** [comment]
> I have minor comments below, but no objection if you want to merge this PR.
> 
> Are these tests disabled right now? I see these messages in the test log:
> 
> ```
> TASK [Test_Execution : Import constants] ***************************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:3
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> META: ran handlers
> 
> TASK [Test_Execution : Create a pod to encapsulate PKI ACME containers] ********
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:10
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Deploy PostgreSQL persistent database containers] *******
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:13
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Create a directory to store database file] **************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:16
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : create file database class file] ************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:22
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : create file database url file] **************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:28
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : create file database user file] *************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:34
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : create file database password file] *********************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:40
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Deploy the PKI ACME responder] **************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:46
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Stop a created pod] *************************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:50
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Remove a created pod] ***********************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_podman.yml:54
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Import constants] ***************************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:3
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> META: ran handlers
> 
> TASK [Test_Execution : Download oc cli binary to access openshift] *************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:10
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : OC login] ***********************************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:16
> skipping: [192.168.33.20] => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}
> 
> TASK [Test_Execution : Install acme on openshift] ******************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:20
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Deploying Persistent Database] **************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:23
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Wait for deploy acme] ***********************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:26
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Uncomment database lines from file] *********************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:30
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Removing InMemoryDatabase entry] ************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:36
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Deploy Database acme configuration] *********************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:42
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : restart acme pod] ***************************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:45
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Wait for restart acme pod] ******************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:48
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : oc get acme route] **************************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:52
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Removing pki-acme app] **********************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:58
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : Removing postgressql database app] **********************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:61
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : removing postgressql secret] ****************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:65
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : removing postgressql persistent volume] *****************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:68
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> 
> TASK [Test_Execution : logging out from openshift] *****************************
> task path: /vagrant/tests/dogtag/pytest-ansible/installation/roles/Test_Execution/tasks/configure_acme_oc.yml:71
> skipping: [192.168.33.20] => {"changed": false, "skip_reason": "Conditional result was False"}
> ```

This is because here it's only executing operation related to topology name topology-02 but for acme we are using topology name acme, I have added the .gitlab-ci.yml  entry for acme test which would be run in CI pipeline. Same test is working in downstream. Thanks

**Comment by 06shalini** [comment]
@dpuniaredhat 
You could  try running the job in pipeline the way it is done in following PR :
https://github.com/dogtagpki/pki/pull/3351/files

You need to edit following files:
 .github/workflows/qe-tests.yml
 ci/Vagrantfile
 tests/dogtag/pytest-ansible/requirements.txt 

The changes related to acme run can be reverted before merge of this PR.

This comment is not a mandatory comment for existing tests, we need to follow this for any new test we write.

**Review by 06shalini — APPROVED** [review]

**Comment by dpuniaredhat** [comment]
> @dpuniaredhat
> You could try running the job in pipeline the way it is done in following PR :
> https://github.com/dogtagpki/pki/pull/3351/files
> 
> You need to edit following files:
> .github/workflows/qe-tests.yml
> ci/Vagrantfile
> tests/dogtag/pytest-ansible/requirements.txt
> 
> The changes related to acme run can be reverted before merge of this PR.
> 
> This comment is not a mandatory comment for existing tests, we need to follow this for any new test we write.

Executed the test for acme topology and run the test and it's passing. https://github.com/dogtagpki/pki/pull/3352/checks?check_run_id=1252877408

**Pull request merged**